src/main/java/programminglife/model/Graph.java

```diff
@@ -0,0 +1,127 @@
+package programminglife.model;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.*;
+
+/**
+ * Created by marti_000 on 25-4-2017.
+ */
+public class Graph {
+    private String id;
+    private Set<Node> rootNodes;
+    private List<Node> nodes;
+
+    public Graph(String id) {
+        this.nodes = new ArrayList<>();
+        this.nodes.add(new Node(-1));
+        this.id = id;
+        this.rootNodes = new HashSet<>();
+    }
+
+    public boolean addNode(Node node) {
+        return this.nodes.add(node);
+    }
+
+    public Node getNode(int id) {
+        try {
+            return this.nodes.get(id);
+        } catch (IndexOutOfBoundsException e) {
+            return null;
+        }
+    }
+
+    private void parseLink(Scanner sc) {
+        int sourceId = Integer.parseInt(sc.next());
+        sc.next();
+        int destinationId = Integer.parseInt(sc.next());
+        sc.next();
+        String overlap = sc.next();
+
+        Node sourceNode = this.getNode(sourceId);
+        Node destinationNode = this.getNode(destinationId);
+
+        if (null == sourceNode)
+            throw new Error("Source node does not (yet) exist!");
```

Comment:
Because it cannot be added at the correct position in the list. Also, `sourceNode == null` would violate our initial assumption that nodes appear in sequential order and source nodes appear before their respective outlinks.